driver-core/src/test/java/com/datastax/driver/core/RollingCountTest.java

```diff
@@ -0,0 +1,144 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core;
+
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.util.concurrent.TimeUnit;
+
+import static com.datastax.driver.core.Assertions.assertThat;
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.mockito.Mockito.when;
+
+public class RollingCountTest {
+    @Mock
+    com.codahale.metrics.Clock clock;
+
+    RollingCount rollingCount;
+
+    @BeforeMethod(groups = "unit")
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+        assertThat(clock.getTick()).isEqualTo(0);
+        rollingCount = new RollingCount(clock);
+    }
+
+    @Test(groups = "unit")
+    public void should_record_adds_in_first_interval() {
+        rollingCount.increment();
+        setTime(1, SECONDS);
+        rollingCount.increment();
+        setTime(2, SECONDS);
+        rollingCount.increment();
+
+        setTime(6, SECONDS);
+        assertThat(rollingCount.count()).isEqualTo(3);
+    }
+
+    @Test(groups = "unit")
+    public void should_record_adds_over_two_intervals() {
```

Comment:
Nice, another + over EWMA.  Increments after the first interval are treated the same as those in the first, where with EWMA incremented the rate based on some alpha after the first interval.